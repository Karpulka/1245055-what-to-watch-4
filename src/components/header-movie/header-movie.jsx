import React from "react";
import {AuthorizationStatus} from "../../reducer/user/user";
import {getAuthorizationStatus} from "../../reducer/user/selectors";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import Breadcrumbs from "../breadcrumbs/breadcrumbs.jsx";

const HeaderMovie = (props) => {
  const {auth, isBreadcrumbs} = props;

  return <React.Fragment>
    {isBreadcrumbs ? <Breadcrumbs /> : ``}
    <div className="user-block">
      {auth === AuthorizationStatus.AUTH ? <div className="user-block__avatar">
        <img src="img/avatar.jpg" alt="User avatar" width="63" height="63"/>
      </div> : <a href="/login" className="user-block__link">Sign in</a>}
    </div>
  </React.Fragment>;
};

HeaderMovie.propTypes = {
  auth: PropTypes.string.isRequired,
  isBreadcrumbs: PropTypes.bool
};

const mapStateToProps = (state) => ({
  auth: getAuthorizationStatus(state)
});

export {HeaderMovie};
export default connect(mapStateToProps)(HeaderMovie);
