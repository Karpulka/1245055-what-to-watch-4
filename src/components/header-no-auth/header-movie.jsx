import React from "react";
import {AuthorizationStatus} from "../../reducer/user/user";
import {getAuthorizationStatus} from "../../reducer/user/selectors";
import PropTypes from "prop-types";
import {connect} from "react-redux";

const HeaderMovie = (props) => {
  const {isAuth} = props;

  return <div className="user-block">
    {isAuth === AuthorizationStatus.AUTH ? <div className="user-block__avatar">
      <img src="img/avatar.jpg" alt="User avatar" width="63" height="63"/>
    </div> : <a href="/login" className="user-block__link">Sign in</a>}
  </div>;
};

HeaderMovie.propTypes = {
  isAuth: PropTypes.string.isRequired
};

const mapStateToProps = (state) => ({
  isAuth: getAuthorizationStatus(state)
});

export {HeaderMovie};
export default connect(mapStateToProps)(HeaderMovie);
