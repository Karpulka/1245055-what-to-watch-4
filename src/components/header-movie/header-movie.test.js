import React from "react";
import renderer from "react-test-renderer";
import {HeaderMovie} from "./header-movie";

it(`Render HeaderMovie no Auth`, () => {
  const tree = renderer
    .create(<HeaderMovie auth={`NO_AUTH`} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render HeaderMovie Auth`, () => {
  const tree = renderer
    .create(<HeaderMovie auth={`AUTH`} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
