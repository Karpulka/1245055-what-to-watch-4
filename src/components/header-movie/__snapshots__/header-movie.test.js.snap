// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render HeaderMovie Auth 1`] = `
Array [
  "",
  <div
    className="user-block"
  >
    <div
      className="user-block__avatar"
    >
      <img
        alt="User avatar"
        height="63"
        src="img/avatar.jpg"
        width="63"
      />
    </div>
  </div>,
]
`;

exports[`Render HeaderMovie no Auth 1`] = `
Array [
  "",
  <div
    className="user-block"
  >
    <a
      className="user-block__link"
      href="/login"
    >
      Sign in
    </a>
  </div>,
]
`;
